src/test/java/reposense/authorship/AnnotatorAnalyzerTest.java

```diff
@@ -0,0 +1,79 @@
+package reposense.authorship;
+
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import reposense.authorship.model.FileInfo;
+import reposense.authorship.model.FileResult;
+import reposense.authorship.model.LineInfo;
+import reposense.model.Author;
+import reposense.parser.InvalidLocationException;
+import reposense.template.GitTestTemplate;
+import reposense.util.TestUtil;
+
+public class AnnotatorAnalyzerTest extends GitTestTemplate {
+
+    private static final Author MAIN_AUTHOR = new Author(MAIN_AUTHOR_NAME);
+    private static final Author FAKE_AUTHOR = new Author(FAKE_AUTHOR_NAME);
+    private static final Date SINCE_DATE = TestUtil.getSinceDate(2018, Calendar.FEBRUARY, 8);
+    private static final Date UNTIL_DATE = TestUtil.getUntilDate(2018, Calendar.FEBRUARY, 10);
+
+    public FileResult getFileResult(String relativePath) {
+        FileInfo fileInfo = FileInfoExtractor.generateFileInfo(config.getRepoRoot(), relativePath);
+        return FileInfoAnalyzer.analyzeFile(config, fileInfo);
+    }
+
+    /**
+     * Asserts the correctness of authorship overriding with regards to @@author tags given
+     * in the test file.
+     */
+    public void assertAnnotationAnalysisCorrectness(FileResult fileResult, Author originalAuthor,
+            Author overridingAuthor) {
+        for (LineInfo line : fileResult.getLines()) {
+            Author lineAuthor = line.getAuthor();
+            if (line.getContent().startsWith("fake")) {
+                Assert.assertEquals(originalAuthor, lineAuthor);
+            } else {
+                Assert.assertEquals(overridingAuthor, lineAuthor);
+            }
+        }
+    }
```

Comment:
These two can be put below the test cases

Reply:
Done.